blocks/search/search.js

```diff
@@ -183,63 +220,89 @@ function searchBox(block, config) {
-  const searchResults = searchResultsContainer(block);
-  resultsContainer.appendChild(searchResults);
+  const resultsContainer = createSearchResultsContainer();
 
   searchInput.append(searchIconInInput);
   inputWrapper.append(searchInput);
 
   box.append(inputWrapper, toggleButton, resultsContainer);
 
+  /**
+   * Clicking the search icon toggles the expanded search field.
+   */
   toggleButton.addEventListener('click', () => {
     box.classList.toggle('search__box--expanded');
     toggleButton.toggleAttribute('aria-expanded');
     if (box.classList.contains('search__box--expanded')) {
       searchInput.focus();
     } else {
-      searchInput.value = '';
       clearSearch(block);
     }
   });
 
-  searchInput.addEventListener('input', (e) => {
-    handleSearch(e, block, config);
-  });
-
-  searchInput.addEventListener('keyup', (e) => {
+  /**
+   * Search after input into search field.
+   */
+  const debouncedHandleSearch = debounce((e) => handleSearch(e?.target, block, config), 200);
+  searchInput.addEventListener('input', debouncedHandleSearch);
+
+  /**
+   * Kick off search if search field already has a value when it gains focus.
+   */
+  searchInput.addEventListener('focus', (e) => handleSearch(e?.target, block, config));
+
+  /**
+   * Handle escape being pressed on input or when focused on a result.
+   */
+  block.addEventListener('keyup', (e) => {
+    // Collapse and clear search after pressing Escape.
     if (e.code === 'Escape') {
       box.classList.remove('search__box--expanded');
       toggleButton.toggleAttribute('aria-expanded');
-      searchInput.value = '';
       clearSearch(block);
     }
-    // if (e.code === 'Enter') {
-    //   window.location.href = `/search-results?q=${encodeURIComponent(searchInput.value)}`;
-    // }
   });
 
+  /**
+   * Collapse and clear search after clicking somewhere else.
+   */
   document.addEventListener('click', (e) => {
-    if (!box.contains(e.target) && box.classList.contains('search__box--expanded')) {
+    const collapseAndClear = () => {
       box.classList.remove('search__box--expanded');
       clearSearch(block);
+    };
+
+    if (box.closest('.nav')?.classList.contains('nav--large-screens')) {
+      if (!box.contains(e.target) && box.classList.contains('search__box--expanded')) {
+        collapseAndClear();
+      }
+    } else {
+      if (inputWrapper !== e.target && !inputWrapper.contains(e.target) && resultsContainer !== e.target && !resultsContainer.contains(e.target)) {
+        collapseAndClear();
+      }
     }
   });
 
   return box;
 }
 
+/**
+ * Search block
+ */
 export default async function decorate(block) {
-  const placeholders = await fetchPlaceholders();
-  //!! TODO: handle search functionality
-  const source = block.querySelector('a[href]') ? block.querySelector('a[href]').href : './sample-search-data/query-index.json';
+  // Placeholder strings; not currently in use.
+  const placeholders = {};
```

Comment:
Goodbye, old friend. (Thanks for getting rid of this.)

<img width="514" alt="Screenshot 2025-06-23 at 9 30 42 PM" src="https://github.com/user-attachments/assets/8309a21a-4cb2-4c25-902a-ca4d4fffb84b" />